agent/consul/acl_test.go

```diff
@@ -319,14 +319,16 @@ func testPolicyForID(policyID string) (bool, *structs.ACLPolicy, error) {
 			RaftIndex:   structs.RaftIndex{CreateIndex: 1, ModifyIndex: 2},
 		}, nil
 	case "acl-wr":
-		return true, &structs.ACLPolicy{
+		p := &structs.ACLPolicy{
```

Comment:
How come this change was only needed for a subset of the cases?

Reply:
I believe it is because only the "Concurrent-Token-Resolve" test cases  has a race on `SetHash`, and only these two cases are used by the "Concurrent-Token-Resolve" test case.

Reply:
I think adding it to all of them would be nice for that reason. Would also avoid that confusion when someone looks at this again later.

Reply:
Sounds good! Added those in the latest commit